src/dapp/walletProvider.ts

```diff
@@ -134,7 +134,9 @@ export class WalletProvider implements IDappProvider {
 
             window.addEventListener('message', connectUrl);
         }).then((connectionUrl: string) => {
-            window.location.href = `${this.baseWalletUrl()}${connectionUrl}?callbackUrl=${options !== undefined && options.callbackUrl !== undefined ? options.callbackUrl : window.location.href}`;
+            const callbackUrl = `callbackUrl=${options !== undefined && options.callbackUrl !== undefined ? options.callbackUrl : window.location.href}`;
+            const token = options !== undefined && options.token !== undefined ? `&token=${options.token}` : '';
+            window.location.href = `${this.baseWalletUrl()}${connectionUrl}?${callbackUrl}${token}`;
```

Comment:
I think if we do it something like this it would be even more readable:

```js
let callbackUrl = `callbackUrl=${window.location.href}`;
if (options && options.callbackUrl) {
  callbackUrl = `callbackUrl=${options.callbackUrl}`;
}

let token = '';
if (options && options.token) {
  token = `&token=${options.token}`;
}
```

Reply:
OK